boards/xtensa/esp32s2/esp32s2_defconfig

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_XTENSA_RESET_VECTOR=n
+
+CONFIG_BOARD_ESP32S2=y
+CONFIG_SOC_ESP32S2=y
+
+CONFIG_MAIN_STACK_SIZE=2048
```

Comment:
What special requirements are needed to specify the main stack size in the board defconfig file?

Reply:
Copied from ESP32's board def; so it comes from there. Seems like alot of boards do this even though its probably best if they didnt; but that might be a good change in another PR